### PR TITLE
Documentation: Improve the way CHANGELOG files are maintained

### PR DESCRIPTION
**Pull request by gziolo**
## Description

This PR aims to improve the way CHANGELOG files and version bumps are maintained for all npm packages. The changes proposed include:
1. All packages will have `CHANGELOG.md` file that starts with:
    ```md
    <!-- Learn how to maintain this file at https://github.com/WordPress/gutenberg/tree/master/packages#maintaining-changelogs. -->
    
    ## Unreleased
    ```
    The updated title (`Unreleased`) follows the recommendation shared at https://keepachangelog.com/en/1.0.0/:
    > Keep an Unreleased section at the top to track upcoming changes.
    > This serves two purposes:
    > - People can see what changes they might expect in upcoming releases
    > - At release time, you can move the Unreleased section changes into a new release version section.
    
    We follow that process now.
2. The recommended version for the newly created package becomes `1.0.0-pre` to leave the room for Lerna to bump to `1.0.0`.

All related docs were updated to reflect these changes.

In addition, the script that automates npm prerelease process was updated to work with the new `## Unreleased` heading. I also included some relaxed checks for the version bump detection to reflect the fact that contributors use lowercase for title of headings.

### CODE REVIEW

**Comment by gziolo** [comment]
Bonus question, they include `# Changelog` title at https://keepachangelog.com/en/1.0.0/. Should we add it to all related files as well?

One more clarification, I opted for using HTML comment when including the link to the documentation that covers maintaining changelog files. The idea is to hide it on GitHub when reading but keep visible when changing.

**Comment by aduth** [comment]
Related: #22003

File: packages/a11y/CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
-## Master
+## Unreleased
+
+<!-- Learn how to maintain this file at https://github.com/WordPress/gutenberg/tree/master/packages#maintaining-changelogs. -->
```
Comment (aduth) [inline packages/a11y/CHANGELOG.md:3]: I assume it should be?

```suggestion
<!-- Learn how to maintain this file at https://github.com/WordPress/gutenberg/tree/master/packages#maintaining-changelogs. -->

## Unreleased
```

File: packages/README.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ When creating a new package, you need to provide at least the following:
     ```json
     {
     	"name": "@wordpress/package-name",
-    	"version": "1.0.0-beta.0",
+    	"version": "1.0.0-pre",
```
Comment (aduth) [inline packages/README.md:13]: Kinda wonder if we should be more clear here with

```suggestion
    	"version": "1.0.0-prerelease",
```

**Review by aduth — APPROVED** [review]

**Pull request merged**